clients/bigquery/cast.go

```diff
@@ -18,8 +18,9 @@ func castColVal(colVal any, colKind columns.Column, additionalDateFmts []string)
 	if colVal == nil {
 		return nil, nil
 	}
-
 	switch colKind.KindDetails.Kind {
+	case typing.Float.Kind, typing.Integer.Kind, typing.Boolean.Kind:
```

Comment:
If we add an explicit case for `typing.String` we can then make the default case throw an error that the column kind is not supported which will catch potential errors if we add more typing kinds.